library/kani_macros/src/contract.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Utility functions for macro expansion of function contract clauses.
+use proc_macro2::TokenStream as TokenStream2;
+use proc_macro2::{Ident, Span};
+use quote::quote;
+use syn::{Attribute, Expr, FnArg, ItemFn, Signature};
+use uuid::Uuid;
+
+/// Given a function `foo`, this function creates a closure with name
+/// `foo_<uuid>`, where <uuid> is a unique identifier.
+/// The closure has the same function arguments and the same function body
+/// as the original function `foo`.
+pub fn convert_to_closure(item: &ItemFn) -> (Ident, TokenStream2) {
+    let fn_sig = &item.sig;
+    let body = &item.block;
+    let args_with_ty = &fn_sig.inputs;
+    let rt = &fn_sig.output;
+    let name = format!("{}_{}", fn_sig.ident.clone().to_string(), Uuid::new_v4()).replace("-", "_");
+    let ident = Ident::new(name.as_str(), Span::call_site());
+    let inner_fn = quote! {
+        let #ident = |#args_with_ty| #rt #body;
+    };
+    (ident, inner_fn)
+}
+
+/// Converts all "#[kani::ensures(...)]" attributes to "kani::postcondition(...)" statement tokens.
+pub fn extract_ensures_as_postconditions(attributes: &Vec<Attribute>) -> TokenStream2 {
+    attributes
+        .iter()
+        .filter_map(|a| {
+            let name = a.path.segments.last().unwrap().ident.to_string();
```

Comment:
If you only look at the last part, wuoldn't that match `whatever::ensures` too?

Reply:
Thanks for pointing it out. I've added a helper function to deal with this now.